sdk/go/common/resource/config/map.go

```diff
@@ -28,29 +28,76 @@ type Map map[Key]Value
 
 // Decrypt returns the configuration as a map from module member to decrypted value.
 func (m Map) Decrypt(decrypter Decrypter) (map[Key]string, error) {
-	r := map[Key]string{}
-	for k, c := range m {
-		v, err := c.Value(decrypter)
+	ctx := context.TODO()
+
+	if decrypter == NopDecrypter {
+		result := map[Key]string{}
+		for k, v := range m {
+			vv, err := v.Value(decrypter)
+			if err != nil {
+				return nil, err
+			}
+			result[k] = vv
+		}
+		return result, nil
+	}
+
+	objectMap := map[Key]object{}
+	for k, v := range m {
+		obj, err := v.coerceObject()
```

Comment:
not sure that we want to do coercion here--guessing that we probably want values as-is. was something broken without this?